### PR TITLE
Annotate and move Tagalong to sprites

### PR DESCRIPTION
**Pull request by scawful**
- Fills in missing data tables
- Replaces all anonymous branching labels and local data addresses with labels 
- Adds logical indentation for comparison statements 
- Unused old man code marked with `\hacker` for potential expansion

### CODE REVIEW

File: sprites/tagalong.asm [hunk 1]
```diff
@@ -0,0 +1,1614 @@
...
+    LDA $1A00, X : STA $00
+    LDA $1A14, X : STA $01
+    
+    LDA $1A28, X : STA $02
+    LDA $1A3C, X : STA $03
+    
+    STZ $0B
+    
+    LDA $1A50, X : STA $0A : BPL .non_negative_altitude
+      LDA.b #$FF : STA $0B
```
Comment (JaredBrian) [inline sprites/tagalong.asm]: 2 space format 😏

File: sprites/tagalong.asm [hunk 2]
```diff
@@ -0,0 +1,1614 @@
...
+  .non_negative_altitude
+    
+    REP #$20
+    
+    LDA $00 : CLC : ADC $0A : CLC : ADC.w #$000C : STA $00
+    LDA $02 : CLC : ADC.w #$0008 : STA $02
+    
+    LDA.w #$1568 : SEC : SBC $00 : BPL .non_negative_delta_x
+      EOR.w #$FFFF : INC A
+  .non_negative_delta_x
```
Comment (JaredBrian) [inline sprites/tagalong.asm]: labels should be on the same indentation of their opening "if" statement or loop
pretending they are the {} of a regular if
ex:
```CSharp
if ()
{
    code ...
}
```

```
LDA.b $00 : CMP.b #$01 : BNE .not1
    code ...

.not1

File: sprites/tagalong.asm [hunk 3]
```diff
@@ -0,0 +1,1614 @@
...
+Tagalong_MessageTagalongs:
+    db $05 ; Zelda's rescue telepathy 
+    db $09 ; Locksmith 
+    db $0A ; Kiki
+    
+Tagalong_MessageTimers:
+    dw $0DF3, $06F9, $0DF3
+  
+Tagalong_MessageIds:
+    dw $0020, $0180, $011D
```
Comment (JaredBrian) [inline sprites/tagalong.asm:200]: data blocks should have {} too

File: sprites/tagalong.asm [hunk 4]
```diff
@@ -0,0 +1,1614 @@
...
+    .just_draw
+    
+    BRL Tagalong_Draw
+    
+    RTS ; Unreachable
+}
+
+; ==============================================================================
+
+; \scawful: Potential expansion point for the Tagalong system.
```
Comment (JaredBrian) [inline sprites/tagalong.asm]: i like this idea of marking potential expansion places. would it make sense to mark it as \hacker or something?

File: sprites/tagalong.asm [hunk 5]
```diff
@@ -0,0 +1,1614 @@
...
+    LDA.w Tagalong_TriggerData_area_data_boundaries_1, X : TAX
+    
+    .next_area_data_block
+    
+    STX $0C
+    STZ $0A
+    
+    LDA $7EF3CC : AND.w #$00FF : CMP $A55C, X : BNE .not_area_data_match
+    
+    LDA.w Tagalong_TriggerData_area_data_1, X : STA $00
```
Comment (JaredBrian) [inline sprites/tagalong.asm:980]: these are wrong i'm pretty sure.
 it should be +0, +2, +4, and +6 not +0, +2, +6, +8
 
 ```
LDA.w $A554, X : STA $00
LDA.w $A556, X : STA $02
LDA.w $A558, X : STA $06
LDA.w $A55A, X : STA $04
```

File: sprites/tagalong.asm [hunk 6]
```diff
@@ -0,0 +1,1614 @@
...
+    
+    RTS
+}
+
+; ==============================================================================
+
+; $04A6CD-$04A906 DATA
+TagalongDraw_Drawing:
+{    
+  .props
```
Comment (JaredBrian) [inline sprites/tagalong.asm]: labels for data should be on the same indentation as the data as well

Comment (JaredBrian) [reply sprites/tagalong.asm]:
```
    .props
    db $20, $C0, $00 ; up
    db $00, $A0, $00 ; down
    db $40, $60, $00 ; left
    db $40, $60, $44 ; right
```

File: sprites/tagalong.asm [hunk 7]
```diff
@@ -0,0 +1,1614 @@
...
+Tagalong_AnimateMovement:
+    STA $05 : AND.b #$20 : LSR #2 : TAY
+    
+    LDA $05 : AND.b #$03 : STA $04
+    
+    STZ $72
+    
+    CPY.b #$08 : BNE .low_priority
+      LDY.b #$00
+      LDA   $7EF3CC
```
Comment (JaredBrian) [inline sprites/tagalong.asm]: whats with the random white space? lol
```
LDA    $7EF3CC
```

File: sprites/tagalong.asm (hunk 3 above)
Comment (scawful) [reply sprites/tagalong.asm:200]: Done.

File: sprites/tagalong.asm (hunk 5 above)
Comment (scawful) [reply sprites/tagalong.asm:980]: Done.

File: sprites/tagalong.asm (hunk 4 above)
Comment (scawful) [reply sprites/tagalong.asm]: done

File: sprites/tagalong.asm (hunk 1 above)
Comment (scawful) [reply sprites/tagalong.asm]: Genuinely looks terrible but done

File: sprites/tagalong.asm (hunk 2 above)
Comment (scawful) [reply sprites/tagalong.asm]: Done

File: sprites/tagalong.asm (hunk 6 above)
Comment (scawful) [reply sprites/tagalong.asm]: Done

File: sprites/tagalong.asm (hunk 7 above)
Comment (scawful) [reply sprites/tagalong.asm]: done

File: sprites/tagalong.asm (hunk 1 above)
Comment (JaredBrian) [reply sprites/tagalong.asm]: 🤣 thanks

**Review by JaredBrian — APPROVED** [review]

**Pull request merged**